src/libraries/System.Security.Principal.Windows/src/System/Security/Principal/WindowsIdentity.cs

```diff
@@ -677,6 +678,11 @@ public static T RunImpersonated<T>(SafeAccessTokenHandle safeAccessTokenHandle,
             return result;
         }
 
+        public static Task RunImpersonatedAsync(SafeAccessTokenHandle safeAccessTokenHandle, Func<Task> func)
+            => RunImpersonated(safeAccessTokenHandle, func);
+
+        public static Task<T> RunImpersonatedAsync<T>(SafeAccessTokenHandle safeAccessTokenHandle, Func<Task<T>> func)
```

Comment:
Nit: can you add XML comments to these two methods?  They'll be used to seed the docs.  You can comment the contents of the current docs for the sync functions and then modify them appropriately.

Reply:
will do

Reply:
@stephentoub are nested generics supported in comments?I cannot find an example con codebase and as is I get a CS0081 error.

Reply:
Nope.

Reply:
Ok updated with "plain" type name, let me know if you know better way.